src/plugins/discover/public/application/angular/discover.js

```diff
@@ -508,7 +512,16 @@ function discoverController($route, $scope) {
       $scope,
       fetch$,
       {
-        next: $scope.fetch,
+        next: async () => {
+          try {
+            await $scope.fetch();
```

Comment:
You could actually make it nicer with

$scope.fetch$().pipe(
    finally(autoRefreshDoneCb)
)

Reply:
I am missing something.
There is no `$scope.fetch$` 
there is only $scope.fetch() and it returns a promise

Reply:
Oh, sorry.
I see that `fetch` still doesn't use searchSource.fetch$ internally.